atoma-proxy/src/telemetry.rs

```diff
@@ -78,13 +80,24 @@ fn init_traces(otlp_endpoint: &str) -> Result<sdktrace::Tracer> {
 pub fn setup_logging<P: AsRef<Path>>(log_dir: P) -> Result<(WorkerGuard, WorkerGuard)> {
     let otlp_endpoint = std::env::var("OTEL_EXPORTER_OTLP_ENDPOINT")
         .unwrap_or_else(|_| DEFAULT_OTLP_ENDPOINT.to_string());
+    let loki_endpoint =
+        std::env::var("LOKI_ENDPOINT").unwrap_or_else(|_| DEFAULT_LOKI_ENDPOINT.to_string());
     // Create logs directory if it doesn't exist
     std::fs::create_dir_all(&log_dir).context("Failed to create logs directory")?;
 
     // Set up metrics
     let metrics_provider = init_metrics(&otlp_endpoint);
     global::set_meter_provider(metrics_provider);
 
+    let (layer, task) = tracing_loki::builder()
+        .label("service_name", "atoma-proxy")?
+        .extra_field("pid", format!("{}", process::id()))?
+        .build_url(Url::parse(&loki_endpoint).unwrap())?;
+
+    // The background task needs to be spawned so the logs actually get
+    // delivered.
+    tokio::spawn(task);
```

Comment:
In that case, we should keep the join handle for later appropriate resource cleaning